File: leetcode/0553-optimal-division/solution.go
import "strings"
func optimalDivision(nums []int) string {
    var strs []string
    for _, x := range nums {
        strs = append(strs, strconv.Itoa(x))
    }
    if len(strs) < 3 {
        return strings.Join(strs, "/")
    }
    return strs[0]+"/("+strings.Join(strs[1:], "/")+")"
}

File: leetcode/0553-optimal-division/solution.rs
impl Solution {
    pub fn optimal_division(nums: Vec<i32>) -> String {
        let nums: Vec<String> = nums.iter().map(i32::to_string).collect();
        if nums.len() < 3 {
            return nums.join("/");
        }
        let mut ret = String::new();
        ret += &nums[0];
        ret += "/(";
        ret += &nums.into_iter()
            .skip(1)
            .collect::<Vec<String>>()
            .join("/");
        ret += ")";
        ret
    }
}
